pkg/ipfix/ipfix_record.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipfix
+
+import (
+	"bytes"
+
+	ipfixentities "github.com/vmware/go-ipfix/pkg/entities"
+)
+
+var _ IPFIXRecord = new(ipfixRecord)
+
+// IPFIXRecord interface is added to facilitate unit testing without involving the code from go-ipfix library.
+type IPFIXRecord interface {
```

Comment:
Let's add only the methods that are used in Antrea. Other methods are not necessary.

Reply:
As long as I don't add all the methods, it will report error like:
`cannot use mockRecord (type *"github.com/vmware-tanzu/antrea/pkg/ipfix/testing".MockIPFIXRecord) as type entities.Record in field value:
    *"github.com/vmware-tanzu/antrea/pkg/ipfix/testing".MockIPFIXRecord does not implement entities.Record (missing AddInfoElement method)`
I have looked into the source code and compared to other mock file. Still struggling to find out the cause... Will continue to work on it

Reply:
Not sure if you have discussed this: "github.com/vmware/go-ipfix/pkg/entities/testing" already has `MockRecord` which is the mock implementation of `Record`. could we use it and avoid adding another interface and mock here?

Reply:
When I added this mock, I was referencing to the IPFIXSet interface, which also have a mock in go-ipfix library. But I'm not sure is there any consideration for us to create these mocks in Antrea? @srikartati

Reply:
> As long as I don't add all the methods, it will report error like:
> `cannot use mockRecord (type *"github.com/vmware-tanzu/antrea/pkg/ipfix/testing".MockIPFIXRecord) as type entities.Record in field value: *"github.com/vmware-tanzu/antrea/pkg/ipfix/testing".MockIPFIXRecord does not implement entities.Record (missing AddInfoElement method)`
> I have looked into the source code and compared to other mock file. Still struggling to find out the cause... Will continue to work on it

Discussed offline. We'll add all the methods into the interface to avoid type mismatch between AggregationFlowRecord.Record and mockRecord in the test case declaration.

Reply:
> When I added this mock, I was referencing to the IPFIXSet interface, which also have a mock in go-ipfix library. But I'm not sure is there any consideration for us to create these mocks in Antrea? @srikartati

Thanks, Quan for the suggestion. 
Anlan, I think it's a good idea. Please explore if you can reuse the mock from there for unit tests.
Probably we can do this for other ipfix entities too in a future PR? @zyiou

Reply:
Thank you to Quan and Srikar. Definitely will do.

Reply:
> Thanks, Quan for the suggestion.
> Anlan, I think it's a good idea. Please explore if you can reuse the mock from there for unit tests.
> Probably we can do this for other ipfix entities too in a future PR? @zyiou

Sure. I think we can move all the mock to go-ipfix, including exporting process & aggregation process & collecting process. Then if there are any changes in the interface, we can handle corresponding mock directly at go-ipfix side. What do you think?

Reply:
We currently do not have any tests in go-ipfix that require mocks of exporting process, aggregation process, and collecting process in the library. So I am not sure moving the mocks there makes complete sense.